### DISCUSSION TITLE
CMI Rank PlaceHolder Help

### DISCUSSION THREAD

**Discussion by TheWookofWallST** [Zrips/CMI#3775] [Q&A]
**Question: **
So I had a previous thread that was the same subject but Zrips replied and closed it. I did what he said and it did fix the rankup % placeholder but the next rank placeholder doesn't seem to work. 

Here is my Rank config
https://pastebin.com/9fFAfef8

![Capture](https://user-images.githubusercontent.com/68366143/87886463-2a284080-c9eb-11ea-8867-8333b00d9e7e.PNG)
![Captu1re](https://user-images.githubusercontent.com/68366143/87886464-2c8a9a80-c9eb-11ea-847f-d7a2c6696323.PNG)

Plugin List: AdvancedNMotd*, AdvancedRegionMarket, AdvancedSpawners, AnimatedScoreboard, BattleArena, BettingGames, BlepFarming, BloodMoon, ChatBrawl, ChestSort, ClearLag, CMI, CMIPaperLib, CrazyAuctions, DragonSlayer, EcoEnchants, EnjinMinecraftPlugin, EpicRename, FogusCore, GCore, GrapplingHook*, Jobs, mcMMO, MobArena, Multiverse-Core, NerfPhantoms, PlaceholderAPI, QuantumShop, SilkSpawners, SpecializedCrates, Towny, TownyChat, TownyNameUpdater, UltraPermissions, Vault, Votifier, VotingPlugin, WorldEdit, WorldGuard

**Comment by Zrips** [comment]
Can you specify which placeholder exactly doesn't work for you?

**Comment by TheWookofWallST** [comment]
Oh yeah my bad! It is the %cmi_user_nextranks% and I was also trying to use %cmi_user_nextvalidranks% neither show the next rank

**Comment by Zrips** [comment]
Hmmm, will need extra information on top of this. First of all, your ranks.yml file and then information at which rank you have been when you checked those placeholders. And keep in mind that %cmi_user_nextvalidranks% will output something only if next ranks to which you can rankup you actually can at this time, which means that requirements to rankup to those ranks are fulfilled.

**Comment by TheWookofWallST** [comment]
Oh okay yeah so %cmi_user_nextvalidranks% would not be what I am looking for. I was on my alt account when I took the screenshot which means I was the default rank I have set, Peasant. 

[ranks.zip](https://github.com/Zrips/CMI/files/4947940/ranks.zip)

**Comment by Zrips** [comment]
Works for me, with your ranks file, i was properly put into Peasant rank and placeholder correctly outputs Merchant as next rank. Check your current rank with /cmi rankinfo, this will output your current rank with next rank list.

**Comment by TheWookofWallST** [comment]
![Capture](https://user-images.githubusercontent.com/68366143/87944376-4ff62980-ca6d-11ea-932a-25f78740cb42.PNG)

Yeah when I do that command it shows for me too that's weird

**Comment by TheWookofWallST** [comment]
just not in the scoreboard

**Comment by Zrips** [comment]
Try using /papi parse %cmi_user_nextranks% if this returns correct value then it means your scoreboard plugin doesnt process placeholder properly

**Comment by TheWookofWallST** [comment]
when i do /papi parse me %cmi_user_nextranks% and I tried another username as well both return blank in chat

**Comment by Zrips** [comment]
Something is strange with your setup... Works for me. Which version of placeholderAPI do you have? And remind me your server version in general.

**Comment by TheWookofWallST** [comment]
I'm using the latest Paper Spigot and PlaceHolderAPI versions

**Comment by Zrips** [comment]
Just tested with latest paper and placeholderAPI, all works as it should. Test this without any other plugin except CMI and placeholderAPI

**Comment by TheWookofWallST** [comment]
Ok I removed all the plugins and it still outputs a blank line

**Comment by Zrips** [comment]
Then i have no idea whats going on in your situation.  Can you use any CMI placeholder in your papi or scoreboard window?
